src/celengine/vecgl.h

```diff
@@ -131,11 +131,14 @@ translate(T x, T y, T z)
     return Eigen::Transform<T,3,Eigen::Affine>(Eigen::Translation<T,3>(Eigen::Matrix<T,3,1>(x,y,z))).matrix();
 }
 
-template <typename T, typename F, std::enable_if_t<std::is_floating_point_v<F>, bool> = true>
-inline T
-mix(const T &x, const T &y, F a)
-{
-    return x * (static_cast<F>(1.0) - a) + y * a;
+template <typename T,
+          typename U,
+          std::enable_if_t<std::is_same_v<typename T::Scalar, typename U::Scalar> &&
+                           std::is_floating_point_v<typename T::Scalar>, bool> = true>
```

Comment:
i'm afraid your change allows to pass Vector3f and Vector4f or Matrix*f and Vector*f together

Reply:
True but you can't add those to each other, so the compilation would fail anyway.